### PR TITLE
Reports API - Adding RSL fucntionality 

### PR DESCRIPTION
**Pull request by olerichard**
Updating the Reports API to be up to date with the current reports backend. 
Alot of models has been updated and changed to reflect current backend and implement new functionality. 
This changes implements the new RLS functionalitry. 

There will be some breaking changes with this. 
But it should not be anything that would cause serious problems. 
No exsisting fields has been removed or renamed. 
Some type changes has occured that will most likely cause some red flags in existing code.

### CODE REVIEW

File: src/http/apiClients/models/report/RlsConfiguration.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+type BasePositionCondition = {
+    id: string;
+    name: string | null;
+};
+
+enum userTypes {
```
Comment (eskil0312) [inline src/http/apiClients/models/report/RlsConfiguration.ts:6]: userTypes should be PascalCase?

File: src/http/apiClients/models/report/RlsConfiguration.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    Any = 'Any',
+}
+
+enum RoleMemberShipType {
+    Unknown = 'Unknown',
+    Department = 'Department',
+    AdGroup = 'AdGroup',
+    DomainMembership = 'DomainMembership',
+    Account = 'Account',
+}
```
Comment (eskil0312) [inline src/http/apiClients/models/report/RlsConfiguration.ts:26]: Is there any reason why we are using enums here instead of types of strings?

Comment (olerichard) [reply src/http/apiClients/models/report/RlsConfiguration.ts:26]: not really. 
Do we avoid enums in general ?

Comment (odinr) [reply src/http/apiClients/models/report/RlsConfiguration.ts:26]: Enums compile to objects in output, so if you want to use the the value as an "Constant" enum is the way to go.
When using only for intelisense/type hinting use `type SomeThing = 'foo' | 'bar'`

File: src/http/apiClients/models/report/index.ts [hunk 3]
```diff
@@ -4,13 +4,30 @@ import ConfigValidation, { ValidationItem } from './ConfigValidation';
...
+    ReportType,
+    ResourceType,
+    RlsConfiguration,
+    SecurityDetails,
+    SecurityRequirement,
+    SecurityRequirementsPolicy,
+    SecurityType,
+    UpdateMarkdown,
+    ValidationItem,
+    TokenType,
```
Comment (odinr) [inline src/http/apiClients/models/report/index.ts:32]: *nitpicking*
when not using the import, standard convention is just to export directly (re-export)
`
export {SomeThing, SomeOtherThing as Foo} from './foo';
export {default as Bar} from './bar'; // @requires Babel^7
export {default} from './default'
`

**Pull request merged**